core/startos/src/service/transition/mod.rs

```diff
@@ -59,21 +61,23 @@ impl Drop for TransitionState {
 }
 
 #[derive(Debug, Clone)]
-pub struct TempDesiredState(pub(super) Arc<watch::Sender<ServiceState>>);
-impl TempDesiredState {
+pub struct TempDesiredRestore(pub(super) Arc<watch::Sender<ServiceState>>, StartStop);
+impl TempDesiredRestore {
     pub fn new(state: &Arc<watch::Sender<ServiceState>>) -> Self {
-        Self(state.clone())
+        Self(state.clone(), state.borrow().desired_state)
```

Comment:
I'm not sure why this was necessary

Reply:
See [transition/mod.rs:74](https://github.com/Start9Labs/start-os/pull/2613/files/a66dd7ea8b2b610bafd315e7e783d09142af3bab#diff-0da36b27c4a67d5ae5b0e098833f4c33b500d8930124b96394352523dc51f062R74)
![image](https://github.com/Start9Labs/start-os/assets/2364004/b5d327f4-fe4b-41bb-8773-74a1f54774e9)